src/openforms/formio/dynamic_config/referentielijsten.py

```diff
@@ -55,7 +58,15 @@ def fetch_options_from_referentielijsten(
 
     try:
         with build_client(service, client_factory=ReferentielijstenClient) as client:
-            result = client.get_items_for_tabel_cached(code)
+            # check if the table is valid (we don't want to show the possible valid
+            # options of an invalid table)
+            table_data = client.get_tabel(code)
+            if table_data:
+                table = Tabel.from_dict(table_data)
```

Comment:
Why not let the client return the data classes directly? You now have a data class and a typed dict describing the data, and they're practically the same. It makes much more sense to only use the dataclasses and not expose the typed dicts at all.

Make `client.get_table` return the `Table` data class instance, instead of converting the dict data outside of the client into such an instance. This will make it so much easier to work with the data and you'll have proper strong type checking.

And please, replace `tabel` with `table` in code.